### ISSUE TITLE
CouchPotato reporting OpenSSL 0.13 is installed when 0.15 is actually installed.

### ISSUE THREAD

**Issue by bonebox** [RuudBurger/CouchPotatoServer#5755]
This error is thrown even though I have OpenSSL 0.15.1 installed.  I've restarted CP several times since upgrading to 0.15.1 from 0.13.1.  CP is obviously still referencing the old version somehow.  What path is CP using to launch Python?

```
BoneStation> pip freeze | grep -i openssl
pyOpenSSL==0.15.1
```

```
BoneStation> python
Python 2.7.9 (default, Apr 21 2015, 03:26:30)
[GCC 4.6.4] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import OpenSSL
>>> print OpenSSL.__version__
0.15.1
```

### Steps to reproduce:
1. Launch CouchPotato

### Information:
Version of CouchPotato: git:(RuudBurger:CouchPotatoServer master) de141aae (2015-11-17 15:32:28)
Running on: Synology DS414

### Logs:
```
11-30 10:09:25 ERROR [chpotato.core._base._core] OpenSSL installed but 0.15 is needed while 0.13.1 is installed. Run `pip install pyopenssl --upgrade` 
```

**Comment by CouchPotatoBot** [comment]
Thanks for the issue report! Before a real human comes by, please make sure your report has all the below criteria checked

- [x] Fill in all the dots. They're there for a reason
- [x] Post logs, either inline (for smaller logs) or using [Pastebin](http://pastebin.com/)
- [x] Please wrap inline logs inside ``` for readability
- [ ] Enabled debug logging in Settings > Advanced (restart CP, be sure to disable after the bug is fixed)
- [x] Post full version information

Please make sure you also read [contribution guide](https://github.com/RuudBurger/CouchPotatoServer/blob/develop/contributing.md#issues) and followed all the steps. 
Make the title describe your issue. Having "CP not working" or "I get this bug" for 100 issues, isn't really helpful. My master will close issues if there isn't enough information. On a good day he will tag the issue on close with the reason (like `can't reproduce`), but usually he won't, the lazy asshat.

Sometimes my master seems like a grumpy cat and responds with short answers. This isn't (always) because he hates you, but because he's on mobile or busy fixing bugs. If something isn't clear, please let him know. Maybe he can teach me his awesome ways.

Thanks!

**Comment by cscattaglia917** [comment]
I'm also running into this issue and have had no luck.  Upgraded pyopenssl to 0.15.1, like bonebox, restarted cp but still having this issue.  Even deleted the cache folder in .couchpotato to no avail.

Brief log entry:
12-01 12:06:34 ERROR [chpotato.core._base._core] OpenSSL installed but 0.15 is needed while 0.13 is installed. Run `pip install pyopenssl --upgrade`
12-01 12:06:34 DEBUG [chpotato.core._base._core] OpenSSL detected: pyopenssl (0.13) using OpenSSL (OpenSSL 1.0.1e 11 Feb 2013)

**Comment by bonebox** [comment]
Not sure what system you're running CP on, but I was able to get past this error by launching CP from the command-line rather than through DSM on my Synology DS414.  Unfortunately, I'm also getting the following error.  This is using Deluge 1.3.5.

```
12-02 08:46:49 ERROR [o.core.downloaders.deluge] Failed to get all torrents: [SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:581) Traceback (most recent call last):
  File "/volume1/@appstore/couchpotatoserver-custom/var/CouchPotatoServer/couchpotato/core/downloaders/deluge.py", line 270, in get_alltorrents
    self.connect()
  File "/volume1/@appstore/couchpotatoserver-custom/var/CouchPotatoServer/couchpotato/core/downloaders/deluge.py", line 222, in connect
    self.client.connect(self.host, int(self.port), self.username, self.password)
  File "/usr/local/couchpotatoserver-custom/var/CouchPotatoServer/libs/synchronousdeluge/client.py", line 143, in connect
    self.transfer.connect((host, port))
  File "/usr/local/couchpotatoserver-custom/var/CouchPotatoServer/libs/synchronousdeluge/transfer.py", line 22, in connect
    self.conn = ssl.wrap_socket(self.sock, None, None, False, ssl.CERT_NONE, ssl.PROTOCOL_TLSv1)
  File "/usr/local/python/lib/python2.7/ssl.py", line 891, in wrap_socket
    ciphers=ciphers)
  File "/usr/local/python/lib/python2.7/ssl.py", line 566, in __init__
    self.do_handshake()
  File "/usr/local/python/lib/python2.7/ssl.py", line 788, in do_handshake
    self._sslobj.do_handshake()
SSLError: [SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:581)
 
```

**Comment by cscattaglia917** [comment]
I'm on Debian and I always start CP from the command line, but still receive the error unfortunately. :(  I have been considering migrating to CentOS recently anyway, I may take this opportunity to make the switch.. Maybe I'll just disable SSL in the meantime.. Not exactly ideal though. 

Any suggestions Ruudburger? Or anything else you need from us?

Thanks!

**Comment by bonebox** [comment]
What version of Deluge are you running @cscattaglia917?  I got the error with 1.3.5; there's no native package for my Synology and the version I was running I built myself using a cross-compiler which was a pretty big pain so I had been putting off updating.  I finally got a new build of 1.3.12 running last night and I'm no longer getting any errors. So you might try updating to the latest Deluge if you haven't already.

**Comment by cscattaglia917** [comment]
I've been using Couchpotato with SABnzbd, not Deluge. I've been on sab 7.20 for the longest time, but recently upgraded to 8.03beta.  However, CP seems to function fine in the background, I just receive an error when attempting to access the page using HTTPS.  Disabling HTTPS "resolves" the issue obviously.

**Comment by bonebox** [comment]
Ah, I forgot the initial error I reported here isn't related to a specific downloader, sorry about that.

**Comment by cscattaglia917** [comment]
No problem, I still appreciate the response

**Comment by bonebox** [comment]
Turns out the issue was indeed a separate, older Python distribution installed on my NAS that CouchPotato was pointing to.  Updating that to the latest fixed the issue. @cscattaglia917 Not sure if your problem is the same, but you might check for other versions of Python installed on your system.  Since you're starting from command line, you could try launching CP while explicitly pointing to the Python version that you know has an updated OpenSSL.

**Comment by cscattaglia917** [comment]
@bonebox - Yeah, I discovered that when I ran pip install pyopenssl, it actually upgraded the pyopenssl package in a python3.2 install that is present on my system, while CouchPotato uses 2.7.  I installed a pip-2.7 version and used that (pip2.7 install pyopenssl) to install pyopenssl on the correct version of python.  Hope this helps anyone else who might run into this.

I ran these commands to install a different version of pip:
sudo apt-get install python-setuptools
sudo easy_install-2.7 pip

Then installed pyopenssl using pip2.7 like so:
pip2.7 install pyopenssl
